notification.c

```diff
@@ -135,6 +135,28 @@ struct mako_notification *get_notification(struct mako_state *state,
 	return NULL;
 }
 
+void close_group_notifications(struct mako_notification *top_notif,
+	       enum mako_notification_close_reason reason) {
+	struct mako_notification *notif, *tmp;
+
+	struct mako_criteria *notif_criteria = create_criteria_from_notification(
+			top_notif, &top_notif->style.group_criteria_spec);
+
+	wl_list_for_each_safe(notif, tmp, &top_notif->state->notifications, link) {
+		if (notif->id == top_notif->id) {
```

Comment:
The top notification may be ungroupped. In which case we probably want to just close it as you've done.

Why do we need this special case though? Can't we just treat the first notification just like other notifications and close it in this loop?

Reply:
That seemed to cause a segfault in match_criteria when strcmp'ing against app names, I think. Not sure exactly, but it seemed create_criteria_from_notification assumes the notif isn't being free'd before the criteria is free'd.

Reply:
Hmm. Nothing inside the criteria struct should be pointing at the notification, all the strings are copied.

Reply:
You're right, it seems I see the segfault for unrelated reasons.